proto/topo.proto

```diff
@@ -90,6 +92,23 @@ message Config {
   }
 }
 
+message CertificateCfg {
+  oneof config {
+    // self_signed will generate local certificates on the node.
+    SelfSignedCertCfg self_signed = 1;
+    // Additional options will be for loading pregenerated certs.
+    // Also support for CSR and generation workflow.
+  }
+}
+
+message SelfSignedCertCfg {
+  string cert_name = 1;
+  string key_name = 2;
+  uint32 key_size = 3;
+  string common_name = 4;
+  string profile_name = 5;
```

Comment:
Some basic comments for the proto fields would be helpful here

Reply:
added